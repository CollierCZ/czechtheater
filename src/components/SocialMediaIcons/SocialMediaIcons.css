.social-media-icons {
    text-align: left;
    margin: 0;
}

.social-media-icons .social-svg {
    text-align: center;
}

.social-media-icon {
    display: inline-block;
    margin: 0 1rem;
    width: 40px;
    height: 40px;
    z-index:2;
}

.main-header .social-media-icon {
    transition: all ease 0.3s;
}

.main-header .social-media-icon .social-svg {
    color: rgb(255, 255, 255, 0.8);
}

.main-header .social-media-icon .social-svg:hover {
    color: #bbc7cc;
    background: white;
}

/* ==========================================================================
       Media Queries - Smaller than 900px
   ========================================================================== */

@media only screen and (max-width: 900px) {
    .social-media-icon {
        display: inline-block;
        width: 35px;
        height: 35px;
    }
}

/* ==========================================================================
       Media Queries - Smaller than 500px
   ========================================================================== */

@media only screen and (max-width: 500px) {
    .social-media-icon {
        margin: 1rem;
        display: inline-block;
        width: 30px;
        height: 30px;
    }
}
