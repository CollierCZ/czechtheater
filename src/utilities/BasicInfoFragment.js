import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


