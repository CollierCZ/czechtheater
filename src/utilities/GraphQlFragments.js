import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


