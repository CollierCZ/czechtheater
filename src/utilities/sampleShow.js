export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }