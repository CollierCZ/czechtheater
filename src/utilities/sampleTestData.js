export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}