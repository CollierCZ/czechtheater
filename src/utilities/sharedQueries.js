import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`