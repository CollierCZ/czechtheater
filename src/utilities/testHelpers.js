import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}