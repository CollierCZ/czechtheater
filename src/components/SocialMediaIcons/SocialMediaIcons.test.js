import React from "react";
import { simpleShallowRender } from "../../utilities/testHelpers";
import SocialMediaIcons from "./SocialMediaIcons";

describe("SocialMediaIcons", () => {
  it("loads correctly with URL", () => {
    simpleShallowRender(<SocialMediaIcons urls={["https://www.facebook.com/czechtheater/"]} />)
  })
  it("loads correctly when empty", () => {
    simpleShallowRender(<SocialMediaIcons />)
  })
})