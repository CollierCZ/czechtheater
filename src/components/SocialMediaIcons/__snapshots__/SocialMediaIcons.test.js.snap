// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaIcons loads correctly when empty 1`] = `""`;

exports[`SocialMediaIcons loads correctly with URL 1`] = `
<div
  className="social-media-icons"
>
  <SocialIcon
    className="social-media-icon"
    key="https://www.facebook.com/czechtheater/"
    style={
      Object {
        "height": null,
        "width": null,
      }
    }
    url="https://www.facebook.com/czechtheater/"
  />
</div>
`;
